contribute/code.rst

```diff
@@ -0,0 +1,158 @@
+-------------------------------
+
+Please ensure that your changeset answers the following questions in its description.
+
+If you've followed the guidelines for commit messages, you can probably just copy the relevant information that you've already written from the commit message into your changeset. GitHub and GitLab will do this for you in single-commit changesets. Otherwise, a description that says "this is complicated to explain, please read the commit messages" is perfectly fine. We trust you to strike the balance.
+
+What issue does your changeset resolve?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All changes should be linked to an issue report that they resolve, whether they add a feature or fix a problem. Is there an issue report filed on any repository which describes the problem you're fixing or the feature you're adding? Does the report specify which devices the issue occurs on or where the feature applies? Does it provide enough information that someone could look at it and know how to reproduce the problem or test the feature? If not, add this information to the issue report! Then, add a link to the issue report in your changeset description.
+
+For example:
+
+    Fixes https://github.com/ubports/ubuntu-touch/issues/1
+
+How did you test that the change was successful?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All changes require testing to ensure they resolve the issue report they're referencing. On which devices did you test your change, and how did you do that? Do you have any questions about existing functionality that might have broke but you are unable to test? Make sure you include this in your changeset description. If you figured out how to reproduce an issue that has long been a mystery, add that to the issue description as well. It helps QA even if it means duplicate information.
+
+For example:
+
+    I've tested this dialer change on the Nexus 5. It should work on all devices since it's not modifying anything device-specific, but more testing would be appreciated. I had to touch a bit of code in the phone number search area to fix a bug (#53) and I'm not entirely sure how to test that.
+
+Do any changes need to be merged before or after this changeset?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some changes depend on one or more changes before they will work correctly. If this is the case, you should document the dependencies in your changeset descriptions. Please document all sets that must be merged *before* and *after* the current one in a series, if needed.
+
+For example, say you've filed three Merge Requests on GitLab, ``ubports/core/docs!1``, ``ubports/core/code!2``, and ``ubports/core/infrastructure!3``. They must be merged in that order. In that case, your MR descriptions would have something similar to this included:
+
+* In ``ubports/core/docs!1``::
+
+    https://gitlab.com/ubports/core/code!2 and https://gitlab.com/ubports/core/infrastructure!3 depend on this MR.
```

Comment:
It would be better to split these lines, or to use a different formatting style so that they are automatically wrapped in the rendered view. Currently one has to scroll in order to see the end of the line.